packages/typespec-ts/test/integration/multipart.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { assert } from "chai";
+import { resolvePath } from "@typespec/compiler";
+import MultiPartClientFactory, {
+  MultiPartClient,
+  createFile,
+  createFileFromStream
+} from "./generated/payload/multipart/src/index.js";
+import { resolve } from "path";
+import { readFile } from "fs/promises";
+import { fileURLToPath } from "url";
+describe("MultiPartClient Rest Client", () => {
+  let client: MultiPartClient;
+
+  beforeEach(() => {
+    client = MultiPartClientFactory({
+      allowInsecureConnection: true
+    });
+  });
+  const root = resolvePath(fileURLToPath(import.meta.url), "../../../temp");
+  const imgPath = resolve(root, "./assets/image.jpg");
+  const pngPath = resolve(root, "./assets/image.png");
+
+  describe("string + bytes", () => {
+    it("Buffer extends Uint8Array should be allowed", async () => {
+      const file = await readFile(imgPath);
+      const result = await client
+        .path("/multipart/form-data/mixed-parts")
+        .post({
+          contentType: "multipart/form-data",
+          body: {
+            id: "123",
+            profileImage: file
+          }
+        });
+      assert.strictEqual(result.status, "204");
+    });
+    it("createFile could return file objects", async () => {
+      const file = createFile(await readFile(imgPath), "test.jpg");
+      const result = await client
+        .path("/multipart/form-data/mixed-parts")
+        .post({
+          contentType: "multipart/form-data",
+          body: {
+            id: "123",
+            profileImage: file
+          }
+        });
+      assert.strictEqual(result.status, "204");
+    });
+    it("createFileFromStream could return file object", async () => {
+      const content = await readFile(imgPath);
+      const file = createFileFromStream(() => content, "test.jpg");
```

Comment:
what the code would be if we are using node 20?

Reply:
It would be like `const file = new File(["your data here"], "filename.txt", {type: "text/html"});`.

If Node 20 
![image](https://github.com/Azure/autorest.typescript/assets/9943211/b1b53a9f-977c-42ae-8d71-d01acc52a738)
 
If Node 18
![image](https://github.com/Azure/autorest.typescript/assets/9943211/9ec2216f-0605-482e-9769-56a151dc6003)

Reply:
Purview datamap's ci is failing on this.